sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -986,11 +996,13 @@ public boolean advance() throws IOException {
               rawRecord.topic(),
               rawRecord.partition(),
               rawRecord.offset(),
+              consumerSpEL.getEventTimestamp(rawRecord),
               decode(rawRecord.key(), source.spec.getKeyCoder()),
               decode(rawRecord.value(), source.spec.getValueCoder()));
 
           curTimestamp = (source.spec.getTimestampFn() == null)
```

Comment:
`curTimestamp` is different from `KafkaRecord.timestamp` when `setTimestampFn()` is there. 
Example like it's a topic of checkout entries, I can `setTimestampFn()` to extract checkout_timestamp as `curTimestamp`, neither Kafka record timestamp nor current_timestamp.